storage/rawstores.go

```diff
@@ -54,27 +56,53 @@ func createHTTPClient(cfg HTTPClientConfig) *http.Client {
 	return c
 }
 
-// NewDataStore creates a new Data Store with the supplied config.
-func NewDataStore(cfg *Config, metricsScope promutils.Scope) (s *DataStore, err error) {
-	defaultClient := http.DefaultClient
-	defer func() {
-		http.DefaultClient = defaultClient
-	}()
-
-	http.DefaultClient = createHTTPClient(cfg.DefaultHTTPClient)
-
-	var rawStore RawStore
-	if fn, found := stores[cfg.Type]; found {
-		rawStore, err = fn(cfg, metricsScope)
-		if err != nil {
-			return &emptyStore, err
-		}
+type dataStoreMetrics struct {
+	cacheMetrics *cacheMetrics
+	protoMetrics *protoMetrics
+	copyMetrics  *copyMetrics
+	stowMetrics  *stowMetrics
+}
 
-		protoStore := NewDefaultProtobufStore(newCachedRawStore(cfg, rawStore, metricsScope), metricsScope)
-		return NewCompositeDataStore(NewURLPathConstructor(), protoStore), nil
+// newDataStoreMetrics initialises all metrics required for DataStore
+func newDataStoreMetrics(scope promutils.Scope) *dataStoreMetrics {
+	failureTypeOption := labeled.AdditionalLabelsOption{Labels: []string{FailureTypeLabel.String()}}
+	return &dataStoreMetrics{
+		cacheMetrics: &cacheMetrics{
```

Comment:
does it make sense to add metrics constructors, like `NewCacheMetrics` alongside the store constructors in their respective files so that the newDataStoreMetrics can just call those and the metrics initialization code is better co-located with their usage? similar to what we're currently doing with `newCopyMetrics`